File: ai-recommender/export_feedback.py
import json
import os
from prisma import Prisma

async def export_feedback_data():
    print("Connecting to database...")
    prisma = Prisma()
    await prisma.connect()

    print("Fetching feedback data...")
    # Get only the essential fields needed for training
    feedback = await prisma.feedback.find_many(
        select={
            'userId': True,
            'eventId': True,
            'rating': True,
        }
    )

    # Create data directory if it doesn't exist
    data_dir = '../data'
    if not os.path.exists(data_dir):
        os.makedirs(data_dir)

    # Save to JSON file
    output_path = os.path.join(data_dir, 'feedback_data.json')
    with open(output_path, 'w') as f:
        json.dump(feedback, f, indent=2)

    print(f"Exported {len(feedback)} feedback records to {output_path}")

    await prisma.disconnect()

if __name__ == "__main__":
    import asyncio
    asyncio.run(export_feedback_data()) 

File: scripts/export-feedback.ts
const { PrismaClient } = require('@prisma/client')
const fs = require('fs')
const path = require('path')

const prisma = new PrismaClient()

async function exportFeedback() {
  try {
    // Fetch all feedback data
    const feedbacks = await prisma.feedback.findMany({
      select: {
        userId: true,
        eventId: true,
        rating: true,
      },
    })

    // Create the output directory if it doesn't exist
    const outputDir = path.join(process.cwd(), 'data')
    if (!fs.existsSync(outputDir)) {
      fs.mkdirSync(outputDir, { recursive: true })
    }

    // Write to JSON file
    const outputPath = path.join(outputDir, 'feedback_data.json')
    fs.writeFileSync(
      outputPath,
      JSON.stringify(feedbacks, null, 2),
      'utf-8'
    )

    console.log(`Successfully exported ${feedbacks.length} feedback records to ${outputPath}`)
  } catch (error) {
    console.error('Error exporting feedback data:', error)
  } finally {
    await prisma.$disconnect()
  }
}

// Run the export function
exportFeedback() 